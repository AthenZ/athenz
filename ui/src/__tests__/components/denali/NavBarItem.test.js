/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { render } from '@testing-library/react';
import NavBarItem from '../../../components/denali/NavBarItem';

describe('NavBarItem', () => {
    it('should render a basic item', () => {
        const { getByTestId } = render(<NavBarItem>An item</NavBarItem>);
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
    });

    it('should render a link item', () => {
        const { getByTestId } = render(<NavBarItem link>An item</NavBarItem>);
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
        expect(navbarItem).toHaveClass('nav-link');
    });

    it('should render an active item', () => {
        const { getByTestId } = render(<NavBarItem active>An item</NavBarItem>);
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
        expect(navbarItem).toHaveClass('active');
    });

    it('should render an active link item', () => {
        const { getByTestId } = render(
            <NavBarItem active link>
                An item
            </NavBarItem>
        );
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
        expect(navbarItem).toHaveClass('active');
        expect(navbarItem).toHaveClass('nav-link');
    });

    it('should render a right-aligned item', () => {
        const { getByTestId } = render(<NavBarItem right>An item</NavBarItem>);
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
        expect(navbarItem).toHaveClass('flush-right');
    });

    it('should render an item with width', () => {
        const { getByTestId } = render(
            <NavBarItem width='100px'>An item</NavBarItem>
        );
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
        expect(navbarItem).toHaveStyle(`width: 100px`);
    });

    it('should render a complex item', () => {
        const { getByTestId } = render(
            <NavBarItem complex>
                <div className='complex-item'>
                    <div className='nav-title' data-testid='navbar-text'>
                        A complex item
                    </div>
                </div>
            </NavBarItem>
        );
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
    });

    it('should render a complex active link item', () => {
        const { getByTestId } = render(
            <NavBarItem complex active link>
                <div className='complex-item'>
                    <div className='nav-title' data-testid='navbar-text'>
                        A complex active link item
                    </div>
                </div>
            </NavBarItem>
        );
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
        expect(navbarItem).toHaveClass('active');
        expect(navbarItem).toHaveClass('nav-link');
    });

    it('should disable animations', () => {
        const { getByTestId } = render(<NavBarItem noanim>An item</NavBarItem>);
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toMatchSnapshot();
    });

    it('should render with an additional class name', () => {
        const { getByTestId } = render(
            <NavBarItem className='custom-class'>An item</NavBarItem>
        );
        const navbarItem = getByTestId('navbar-item');

        expect(navbarItem).toHaveClass('denali-navbar-item');
        expect(navbarItem).toHaveClass('custom-class');
    });
});
