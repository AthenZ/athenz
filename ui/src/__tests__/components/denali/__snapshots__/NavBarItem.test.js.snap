// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarItem should disable animations 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 simple denali-navbar-item"
  data-testid="navbar-item"
>
  An item
</div>
`;

exports[`NavBarItem should render a basic item 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 simple denali-navbar-item"
  data-testid="navbar-item"
>
  An item
</div>
`;

exports[`NavBarItem should render a complex active link item 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 active nav-link denali-navbar-item"
  data-testid="navbar-item"
>
  <div
    class="complex-item"
  >
    <div
      class="nav-title"
      data-testid="navbar-text"
    >
      A complex active link item
    </div>
  </div>
</div>
`;

exports[`NavBarItem should render a complex item 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 denali-navbar-item"
  data-testid="navbar-item"
>
  <div
    class="complex-item"
  >
    <div
      class="nav-title"
      data-testid="navbar-text"
    >
      A complex item
    </div>
  </div>
</div>
`;

exports[`NavBarItem should render a link item 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 simple nav-link denali-navbar-item"
  data-testid="navbar-item"
>
  An item
</div>
`;

exports[`NavBarItem should render a right-aligned item 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 simple flush-right denali-navbar-item"
  data-testid="navbar-item"
>
  An item
</div>
`;

exports[`NavBarItem should render an active item 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 simple active denali-navbar-item"
  data-testid="navbar-item"
>
  An item
</div>
`;

exports[`NavBarItem should render an active link item 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 simple active nav-link denali-navbar-item"
  data-testid="navbar-item"
>
  An item
</div>
`;

exports[`NavBarItem should render an item with width 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 60px;
  position: relative;
  width: 100px;
}

.emotion-0.simple {
  color: #fff;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.simple.nav-link {
  cursor: pointer;
}

.emotion-0.simple.nav-link:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.simple.active {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0.simple.nav-link.active:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 .nav-title {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  height: 100%;
  padding: 0 12px;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link .nav-title {
  cursor: pointer;
}

.emotion-0.active .nav-title,
.emotion-0 .active.nav-title {
  box-shadow: inset 0 -4px 0 #3697f2;
}

.emotion-0 .nav-title:hover {
  color: rgba(255,255,255,0.75);
}

.emotion-0.nav-link.active .nav-title:hover,
.emotion-0.nav-link .active.nav-title:hover {
  box-shadow: inset 0 -4px 0 rgba(54,151,242,0.75);
}

.emotion-0 img.nav-logo {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
}

.emotion-0 .nav-icon,
.emotion-0:hover .nav-icon {
  width: 24px;
  height: 24px;
}

.emotion-0 .nav-icon path,
.emotion-0:hover .nav-icon path,
.emotion-0 .nav-icon ellipse,
.emotion-0:hover .nav-icon ellipse,
.emotion-0 .nav-icon rect,
.emotion-0:hover .nav-icon rect,
.emotion-0 .nav-icon circle,
.emotion-0:hover .nav-icon circle {
  fill: #fff;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
}

.emotion-0.nav-link:hover .nav-icon path,
.emotion-0.nav-link:hover .nav-icon ellipse,
.emotion-0.nav-link:hover .nav-icon rect,
.emotion-0.nav-link:hover .nav-icon circle {
  fill: rgba(255,255,255,0.75);
}

<div
  class="emotion-0 simple denali-navbar-item"
  data-testid="navbar-item"
>
  An item
</div>
`;
